src/main/java/org/mskcc/oncokb/curation/security/CustomOAuthSuccessHandler.java

```diff
@@ -57,13 +61,33 @@ public void onAuthenticationSuccess(HttpServletRequest request, HttpServletRespo
                 authorities,
                 ((OAuth2AuthenticationToken) authentication).getAuthorizedClientRegistrationId()
             );
-            SecurityContextHolder.getContext().setAuthentication(authenticationWithAuthorities);
-            super.onAuthenticationSuccess(request, response, authenticationWithAuthorities);
-        } else {
-            request.getSession().invalidate();
-            SecurityContextHolder.clearContext();
-            redirectStrategy.sendRedirect(request, response, "/logout");
+
+            // Create custom token for frontend Firebase authentication
+            String email = authenticationWithAuthorities.getPrincipal().getAttribute("email");
+            String firebaseCustomToken = null;
+            try {
+                Map<String, Object> additionalClaims = new HashMap<>();
+                // This claim will be used in Firebase rules for authorization
+                additionalClaims.put(Constants.FIREBASE_AUTHORIZED_CLAIM, true);
+                firebaseCustomToken = FirebaseAuth.getInstance().createCustomTokenAsync(email, additionalClaims).get();
```

Comment:
This is a async method per the naming. Shouldn't wait for it?

Reply:
The `.get()` waits for the token creation to complete and returns the result